Support MacOS

montaguegabe opened the pull request:
This issue fixes #23, a problem apparently appearing on MacOS 10.8 and above. The diff command has been changed as well on Mac. Added instructions on how to install the matching `grep` and `diff` commands, and modified the code to use them.

montaguegabe committed:
Allow for different command names on OS X

montaguegabe committed:
Add Mac instructions to README

sp1thas commented on README.md:
```diff
@@ -32,14 +32,17 @@ This is a simple shell script that can be used to ignore files from dropbox usin
 | wget   | `sudo sh -c "$(wget -qO- https://raw.githubusercontent.com/sp1thas/dropboxignore/master/utils/install.sh)"`   |
 | fetch  | `sudo sh -c "$(fetch -o - https://raw.githubusercontent.com/sp1thas/dropboxignore/master/utils/install.sh)"`  |
 
+### Installation notes for MacOS
```
I propose to add these `brew install` commands directly into the `utils/install.sh` file.

montaguegabe replied:
Sounds good - I forgot that users of dropboxignore are pretty much guaranteed to be developers so they will definitely have homebrew

sp1thas commented:
Thank you @montaguegabe for your contribution. Your documentation-related changes should also be applied in `docs/installation.md` (I know... duplicated docs.. we should find a better approach on this)

montaguegabe commented:
I can update so that the brew commands are part of the install script and modify the docs in both places

montaguegabe committed:
Update style to match

(PyCharm gives warnings for removing quotes around "diff" so I left all the quotes after the equals signs for consistency - feel free to add commit to change further)

montaguegabe commented:
One thing to note actually: at least on my Mac running a `brew install` command will automatically update homebrew, which can sometimes take up to ~3 minutes of somewhat verbose output (maybe even longer if it hasn't been updated in a while). An unwary user might think all this heavy lifting is to install dropboxignore. But I guess we can circumvent the auto-update with the `HOMEBREW_NO_AUTO_UPDATE=1` prefix so maybe I'll do that. I'm not exactly sure what the rationale is behind Homebrew trying to update each usage

montaguegabe committed:
Add brew commands to install.sh

montaguegabe committed:
Update docs: Remove section about MacOS, add smaller note saying brew is needed

montaguegabe commented:
Sorry don't merge this yet - I hadn't tested install.sh and brew doesn't work with sudo

montaguegabe committed:
Allow for alternative brew install locations

montaguegabe committed:
Run brew without sudo

Source: https://unix.stackexchange.com/questions/353206/is-there-a-reverse-sudo

montaguegabe committed:
Fix bug: forgot to add "bin" after the brew prefix

montaguegabe commented:
Ok all set!  I figured out how to drop sudo for the brew commands as per the second answer on https://unix.stackexchange.com/questions/353206/is-there-a-reverse-sudo (I actually tested a mock `brew` command to make sure it wasn't secretly executing with root permissions). Also added a fix for the few people who have brew install packages to a different location.  Have tested the last commit: both the install script and the installed command itself.

sp1thas reviewed (approved):
LGTM! Thanks :)
